modules/imapauth/lib/Auth/Source/MyAuth.php

```diff
@@ -0,0 +1,173 @@
+<?php
+class sspmod_imapauth_Auth_Source_MyAuth extends sspmod_core_Auth_UserPassBase {
+
+	/* The database DSN.
+     * See the documentation for the various database drivers for information about the syntax:
+     *     http://www.php.net/manual/en/pdo.drivers.php
+     */
+	private $dsn;
+
+	/* The database username & password. */
+	private $username;
+	private $password;
+	private $use_rc_database;
+	private $table_name;
+	private $mail_host;
+	private $imap_hostname;
+	private $imap_port;
+	private $imap_security;
+	private $imap_additional_options;
+
+	public function __construct($info, $config) {
+		parent::__construct($info, $config);
+
+	if (isset($config['use_rc_database'])) {
+		if ($config['use_rc_database'] == true) $this->use_rc_database = $config['use_rc_database'];
+	}
+
+		if (!is_string($config['dsn'])) {
+			throw new Exception('Missing or invalid dsn option in config.');
+		}
+		$this->dsn = $config['dsn'];
+
+		if (!is_string($config['table_name'])) {
+			throw new Exception('Missing or invalid table_name option in config.');
+		}
+		$this->table_name = $config['table_name'];
+
+		if (!is_string($config['mail_host'])) {
+			throw new Exception('Missing or invalid mail_host option in config.');
+		}
+		$this->mail_host = $config['mail_host'];
+
+		if (!is_string($config['username'])) {
+			throw new Exception('Missing or invalid username option in config.');
+		}
+		$this->username = $config['username'];
+
+		if (!is_string($config['password'])) {
+			throw new Exception('Missing or invalid password option in config.');
+		}
+		$this->password = $config['password'];
+
+		if (!is_string($config['imap_hostname'])) {
+			throw new Exception('Missing or invalid imap_hostname option in config.');
+		}
+		$this->imap_hostname = $config['imap_hostname'];
+
+		if (!is_string($config['imap_port'])) {
+			throw new Exception('Missing or invalid imap_port option in config.');
+		}
+		$this->imap_port = $config['imap_port'];
+
+		if (!is_string($config['imap_security'])) {
+			throw new Exception('Missing or invalid imap_security option in config.');
+		}
+		$this->imap_security = $config['imap_security'];
+
+		if (is_string($config['imap_additional_options'])) {
+		$this->imap_additional_options = $config['imap_additional_options'];
+		}
+	}
+
+	protected function login($username, $password) {
+
+	//Defaults if there is no database entry
```

Comment:
Also here. We try to enforce PSR-2 these days, using spaces instead of tabs, having the opening braces after the function definition, spaces around `=` sign and things like that. It usually helps if you have an IDE that understands coding standards, so that you can configure it with one coding standard and make it check (and even fix) these kind of things for you.